src/lib/logical_query_plan/abstract_lqp_node.cpp

```diff
@@ -68,6 +68,29 @@ AbstractLQPNode::~AbstractLQPNode() {
   if (_inputs[1]) _inputs[1]->_remove_output_pointer(*this);
 }
 
+size_t AbstractLQPNode::hash() const {
```

Comment:
Could you add a short comment on how you decided for this to be the hash function? Especially the special treatment of validate etc. should be explained.

Reply:
I'll address this after we got a working impl..